src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocSummaryLocationCheck.java

```diff
@@ -0,0 +1,275 @@
+ * <a href="https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html">
+ * Documentation Comment Specification</a> permits leading asterisks on the first line.
+ * For these Javadoc comments:
+ * </p>
+ * <pre>
+ * &#47;** *
+ *   * Some text.
+ *   *&#47;
+ * &#47;************
+ *   * Some text.
+ *   *&#47;
+ * &#47;**           **
+ *   * Some text.
+ *   *&#47;
+ * </pre>
+ * <p>
+ * The HTML documentation will be just "Some text." without any asterisks.
+ * Since these asterisks will not appear in the generated documentation,
+ * they should not be considered as the beginning of the text.
+ * The check will not report a violation for such comments when the {@code location}
+ * property is set to {@code SECOND_LINE}.
+ * </p>
+ * <p>
+ * The Javadoc tool also ignores all extra asterisks after the comment start,
+ * but this check does not. For this Javadoc comment:
+ * </p>
+ * <pre>
+ * &#47;****  **
+ *   * Some text.
+ *   *&#47;
+ * </pre>
+ * <p>
+ * The HTML documentation will be "Some text.", but the check will report a violation here.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">
+ * Tight-HTML Rules</a>.
+ * Default value is {@code false}.
+ * </li>
+ * <li>
+ * Property {@code location} - Specify the policy on placement of the javadoc summary.
+ * Default value is {@code second_line}.
+ * </li>
+ * </ul>
+ * <p>
+ * By default Check validate that the summary is located on the second line:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocSummaryLocationCheck"/&gt;
+ * </pre>
+ * <p>
+ * This setting produces a violation for each multi-line comment starting
+ * on the same line as the initial asterisks:
+ * </p>
+ * <pre>
+ * &#47;** This comment causes a violation because it starts on the first line
+ *   * and spans several lines.
+ *   *&#47;
+ * &#47;**
+ *   * This comment is OK because it starts on the second line.
+ *   *&#47;
+ * &#47;** This comment is OK because it is on the single line. *&#47;
+ *</pre>
+ * <p>
+ * To ensure that summary is located on the first line:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocSummaryLocationCheck"&gt;
+ *   &lt;property name="location" value="FIRST_LINE"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * This setting produces a violation for each comment not
+ * starting on the same line as the initial asterisks:
+ * </p>
+ * <pre>
+ * &#47;** This comment is OK because it starts on the first line.
+ *    * There may be additional text.
+ *    *&#47;
+ * &#47;**
+ *   * This comment causes a violation because it starts on the second line.
+ *   *&#47;
+ * &#47;** This single-line comment also is OK. *&#47;
+ * </pre>
+ *
+ * @since 8.26
+ */
+@StatelessCheck
+public class JavadocSummaryLocationCheck extends AbstractJavadocCheck {
```

Comment:
I am not specifically seeing the reason why this has to be a javadoc check over java?
It is all about if the first word should appear on the same line as `/**` or not.
What does the javadoc parser get?

Reply:
Yes, this can be implemented as a Java check using regular expressions. And this check now uses regular expressions, but only because of the limitations of the parser. Regular expressions are not the best nor fastest way to check style. Since we have a parser for Javadoc, we need to use it. Another thing is that currently the imperfection of the parser does not allow us to completely get rid of regular expressions. But this should be fixed by improving the parser.

Reply:
As we moved a lot from original ideas of Check ... I think it is reasoable to rename it, and even make it non-javadoc Check (as we do need regexp now, and there is NO strict definition of leading asteriks and spaces, and matching javadoc will be a strugle).

 I still think that if we do simple content verification after javadoc identifier and report `Javadoc content that starts with "  " should be places on second line`, will be good for users in 99%, 1% should not use this Check if they are so special about javadoc formatting.